### PR TITLE
Friendships v2

### PR DESCRIPTION
**Pull request by Miguelus373**
### What this PR does
- Implements that, when a user accepts the friendship, both the friendship and inverse_friendship are created
- Fix style issues

### CODE REVIEW

File: app/controllers/friendships_controller.rb
Comment (teekaytech) [inline app/controllers/friendships_controller.rb:38]: To make your code clearer and more effective, move model manipulation into model class. 

Your inverse friendship method should be created in the Friendship model. and might be similar to the snippet below:
```ruby
# Friendship model

def confirm_friend
  self.update_attributes(confirmed: true)
  Friendship.create!(friend_id: self.user_id,
                  user_id: self.friend_id,
                  confirmed: true)
end

```

You can then call the `confirm_friend` method in the update action of your friendship controller


if your `confirm_friend` method is similar to the one above, you should be able to find user friends only by user_id in the friendships table:

```ruby
# User model

has_many :confirmed_friendships, -> { where confirmed: true }, class_name: "Friendship"
has_many :friends, through: :confirmed_friendships
```

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've made laudable progress in this milestone.
However, some changes are required before this project can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [x] Let's assume that our application is having 2 users: user1 & user2, and user1 sends a friend request to user2. When user2 logs in and immediately click on `ALL USERS`, then click on `See Profile` of user1 that sent the request, there occurs this error:



- [x] Although you've written some unit tests, no integration test was found in your project. Please, add the integration tests.
- Remember about unit and **integrations** tests! ([screenshot](https://user-images.githubusercontent.com/26856491/97762355-8af60780-1b08-11eb-8d3c-f7ac4e2c741c.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

- [x] Check my comments below and fix as appropriate.



* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

File: app/controllers/friendships_controller.rb
Comment (happiguru) [reply app/controllers/friendships_controller.rb:38]: Fixed

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi **TEAM**, congratulations on a great job :fireworks: You have fixed all but one of the comments made by the previous TSE :slightly_smiling_face: I will therefore highlight it in the comments below:
- Kindly write the **Integration** test. We could write a test for our controllers and views. For example, writing a test to login in a user

Once this is fixed, kindly submit another code review :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Review by addod19 — APPROVED** [review]
## Status: Approved :trophy: :balloon: 

Hi **TEAM**, congratulations on implementing the **Integration** test :fireworks: Now our project meets all the minimum requirements :100: 

Go ahead and merge into development :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Pull request merged**
